packages/did-ethr-resolver/src/implementations/resolver.ts

```diff
@@ -77,7 +84,18 @@ class Resolver implements IResolver {
     did: string,
     selector?: DocumentSelector,
   ): Promise<IDIDDocument> {
-    const match = did.match(DIDPattern);
+    let match;
```

Comment:
As extracting address from DID is common task I believe it make sense to create function and move it DID package

Reply:
You meant to say `match` `DIDPattern` is common task? If yes, that's a valid point and I will do the needful changes. I would use `AddressOf` method to extract address at places it is not being used. Thanks!